ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -1172,10 +1250,15 @@ void maybeNotifyForChunkCreation(PutChunk chunk) {
      * Notify for the creation of the first chunk. To be called after the overall operation is completed if the overall
      * blob is composite. If no first chunk was put successfully, this will do nothing.
      */
-    void notifyForFirstChunkCreation() {
-      String chunkId = firstChunkIdAndProperties.getFirst().getID();
-      BlobProperties chunkProperties = firstChunkIdAndProperties.getSecond();
-      notificationSystem.onBlobCreated(chunkId, chunkProperties, userMetadata, NotificationBlobType.DataChunk);
+    void maybeNotifyForFirstChunkCreation() {
+      if (indexToChunkIds.get(0) != null) {
+        // reason to check for not null: with encryption in play, there are chances that 2nd chunk's encrypt job callback
```

Comment:
if this can happen even without encryption, I don't think there is any necessity to say that this is _because_ encryption was introduced.

Reply:
yes, I will fix the comment.